app/preprints/-components/submit/file/upload-file/component.ts

```diff
@@ -99,4 +101,11 @@ export default class PreprintUpload extends Component<PreprintUploadArgs> {
         }
         this.args.validate(file);
     }
+
+    @task
+    @waitFor
+    async preUpload(): Promise<void> {
+        this.isUploadFileDisplayed = true;
+    }
```

Comment:
This _shouldn't_ be doing anything that would require it to be async. Unless it's a wild hack to get past the race condition, this should still be a synchronous function, and I presume VSCode is complaining about this being marked as async. Can you check to see if it works without being async? If you're sure it needs to be, we can give it a try, but it doesn't make sense to me if that's the case.